src/components/House/House.tsx

```diff
@@ -1,53 +1,121 @@
 
-    const floors = createFloors(FLOOR_COUNT)
+        const floors = createFloors(FLOOR_COUNT)
 
-    const onStartLift = (floor: number) => {
-        setFloorValueButton(
-            {...floorValueButton, floor: floor, isPressed: true}
-        )
-    }
+        const changeButton = (floor: number) => {
+            return (
+                (queue.find(q => q === floor) !== undefined) ||
+                ((liftInMotion.rightLift && targetLiftRight === floor) ||
+                    ((liftInMotion.leftLift && targetLiftLeft === floor)))
+            )
+        }
+
+        const addQueue = (floor: number) => {
+            if (queue.find(f => f === floor)) return;
+            setQueue(queue.length === 0 || queue[0] === floor ? [floor] : [...queue, floor]);
+        }
+
+        useEffect(() => {
+
+                if (queue.length !== 0 && queue[0] !== undefined) {
+
+                    const differenceR = Math.abs(queue[0] - targetLiftRight)
+                    const differenceL = Math.abs(queue[0] - targetLiftLeft)
+
+                    // удаляет дубляж целей в очереди
+                    if (queue[0] === targetLiftRight || queue[0] === targetLiftLeft) {
+                        setQueue([...queue].slice(1))
+                        setLiftInMotion((prev) => ({...prev, rightLift: false, leftLift: false}))
+                    }
+
+                    // если едет только 1 лифт, тогда запускаем второй
+                    else if ((liftInMotion.leftLift && !liftInMotion.rightLift) ||
+                        (!liftInMotion.leftLift && liftInMotion.rightLift)) {
 
-    const onStopLift = () => {
-        setFloorValueButton(
-            {...floorValueButton, isPressed: false}
+                        if (!liftInMotion.rightLift) {
+                            setLiftInMotion((prev) => ({...prev, rightLift: true}))
+                            setTargetLiftRight(queue[0])
+                            setQueue([...queue].slice(1))
+
+                        } else if (!liftInMotion.leftLift) {
+                            setLiftInMotion((prev) => ({...prev, leftLift: true}))
+                            setTargetLiftLeft(queue[0])
+                            setQueue([...queue].slice(1))
+                        }
+                    }
+
+                    // если оба стоят, то условие с дистанцией едет, который ближе
+                    else if (!liftInMotion.leftLift && !liftInMotion.rightLift) {
+
+                        if (differenceR <= differenceL) {
+                            setLiftInMotion((prev) => ({...prev, rightLift: true}))
+                            setTargetLiftRight(queue[0])
+                            setQueue([...queue].slice(1))
+                        } else {
+                            setLiftInMotion((prev) => ({...prev, leftLift: true}))
+                            setTargetLiftLeft(queue[0])
+                            setQueue([...queue].slice(1))
```

Comment:
Везде повторяются эти 3 строчки. Может объединить их в функцию, чтобы не дублировать?

```suggestion
                            function startLift(liftType: 'leftLift' | 'rightLift') {
                              setLiftInMotion((prev) => ({...prev, [liftType]: true}))
                           liftType === 'leftLift' ?  setTargetLiftLeft(queue[0]) : setTargetLiftRight(queue[0]);
                            setQueue([...queue].slice(1))
                            
                            }
```